cli/cmd/docs/generate.go

```diff
@@ -0,0 +1,193 @@
+package docs
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/fs"
+	"log"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+func GenerateCmd(rootCmd *cobra.Command) *cobra.Command {
+	docsCmd := &cobra.Command{
+		Use:   "generate",
+		Short: "Generate CLI documentation",
+		Args:  cobra.MaximumNArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			dir := "docs/docs/references/CLI/"
+			if len(args) > 0 {
+				dir = args[0]
+			}
+			rootCmd.DisableAutoGenTag = true
+			err := GenMarkdownTree(rootCmd, dir)
+			if err != nil {
+				log.Fatal(err)
+			}
+		},
+	}
+	return docsCmd
+}
+
+func GenMarkdownTree(cmd *cobra.Command, dir string) error {
```

Comment:
1. Make lowercase, not used outside package
2. Add comment before explaining how this works at a high level and with links to relevant Cobra docs